src/controller/CHIPCommissionableNodeController.cpp

```diff
@@ -35,5 +37,39 @@ const Mdns::DiscoveredNodeData * CommissionableNodeController::GetDiscoveredComm
     return GetDiscoveredNode(idx);
 }
 
+CHIP_ERROR CommissionableNodeController::RequestCommissioning(int commissionerNumber)
+{
+#if CHIP_DEVICE_CONFIG_ENABLE_COMMISSIONER_DISCOVERY_CLIENT
+    const Mdns::DiscoveredNodeData * commissioner;
+    for (int i = 0; i < CHIP_DEVICE_CONFIG_MAX_DISCOVERED_NODES; i++)
+    {
+        commissioner = GetDiscoveredCommissioner(i);
+        if (commissioner != nullptr && --commissionerNumber == 0)
+        {
+            break;
+        }
+    }
+    if (commissionerNumber != 0)
+    {
+        return CHIP_ERROR_INVALID_ARGUMENT;
+    }
+
+    // Enter commissioning mode, open commissioning window
+    InitServer();
+
+    // Send User Directed commissioning request
+    chip::Transport::PeerAddress commissionerAddress =
+        chip::Transport::PeerAddress::UDP(commissioner->ipAddress[0], commissioner->port);
+    SendUserDirectedCommissioningRequest(commissionerAddress);
+#else
+    return CHIP_ERROR_NOT_IMPLEMENTED;
+#endif // CHIP_DEVICE_CONFIG_ENABLE_COMMISSIONER_DISCOVERY_CLIENT
+    return CHIP_NO_ERROR;
```

Comment:
The `#else` case here is liable to complain about unreached code on some compilers.  Shouldn't the `return CHIP_NO_ERROR` move into the `#if` body?